packages/inventory/src/containers/ProfileView/index.tsx

```diff
@@ -0,0 +1,253 @@
+};
+
+interface DefaultGeographyItemProp {
+  label: string;
+  value?: string | number | string[] | number[];
+}
+
+/** component that renders Geography Items
+ *
+ * @param props - the component props
+ */
+export const GeographyItem = (props: DefaultGeographyItemProp) => {
+  const { label, value } = props;
+  return (
+    <Col md={24} className="geography-item">
+      <p className="item">
+        {label}: {value}
+      </p>
+    </Col>
+  );
+};
+
+/** component that renders service point list
+ *
+ * @param props - the component props
+ */
+const ServicePointProfile = (props: ServicePointsListTypes) => {
+  const { columns, opensrpBaseURL, geoLevel, service } = props;
+  const { broken, errorMessage, handleBrokenPage } = useHandleBrokenPage();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const dispatch = useDispatch();
+  const location = useLocation();
+  const loc = location.pathname.split('/');
+  const spId = loc.pop() as string;
+
+  const inventoriesArray = useSelector((state) => getInventoriesArray(state));
+  const locationUnitById = useSelector((state) => getLocationUnitById(state, spId));
+  const [locationNodeById] = useSelector((state) =>
+    locationsBySearchSelector(state, { searchQuery: spId, geoLevel })
+  );
+  const [structure] = useSelector((state) =>
+    structuresSelector(state, { searchQuery: spId, isJurisdiction: false })
+  );
+  const trees = useSelector((state) => treesSelector(state, {}));
+
+  useEffect(() => {
+    // get structures, this is the most important call for this page
+    const params = {
+      serverVersion: 0,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      is_jurisdiction: false,
+    };
+    const structuresDispatcher = (locations: LocationUnit[] = []) => {
+      return fetchLocationUnits(locations, false);
+    };
+    loadJurisdictions(
+      structuresDispatcher,
+      opensrpBaseURL,
+      params,
+      {},
+      service,
+      LOCATIONS_GET_ALL_SYNC_ENDPOINT
+    ).catch((err: Error) => handleBrokenPage(err));
+    // get root Jurisdictions so we can later get the trees.
+    const jurisdictionsDispatcher = (locations: LocationUnit[] = []) => {
+      return dispatch(fetchLocationUnits(locations, true));
+    };
+    loadJurisdictions(
+      jurisdictionsDispatcher,
+      opensrpBaseURL,
+      undefined,
+      undefined,
+      service
+    ).catch((err: Error) => sendErrorNotification(err.message));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const serve = new OpenSRPService(`${GET_INVENTORY_BY_SERVICE_POINT}/${spId}`, opensrpBaseURL);
+    serve
+      .list()
+      .then(() => {
+        dispatch(fetchInventories([inventory1, inventory2]));
+        setIsLoading(false);
+      })
+      .catch(() => sendErrorNotification(ERROR_OCCURRED));
+  }, [dispatch, location.pathname, locationUnitById, opensrpBaseURL, spId, locationNodeById]);
```

Comment:
would dispatch here not cause unnecessary rerenders, since its re-defined with every re-render, IMO, the only dependency we need to have here is the `rootlocations`,  if you want to be more by the book you could add the `openSRPBaseURL`